src/app/answers/answer-details/answer-details.component.ts

```diff
@@ -1,38 +1,47 @@
-import { debounceTime, delay, distinctUntilChanged, distinctUntilKeyChanged, endWith, filter, ignoreElements, map, mapTo, scan, share, shareReplay, skip, startWith, subscribeOn, switchMap, take, takeWhile, tap, withLatestFrom } from 'rxjs/operators';
-import { AnswerExtra } from 'src/app/models/answer.extra.model';
-import { AnswerThread } from 'src/app/models/answer.thread.model';
-import { FormDetails } from 'src/app/models/form.info.model';
-import { Form } from 'src/app/models/form.model';
-import { BASE_BUTTON_VARIANTS, Variants } from 'src/app/shared/base-button/base-button.component';
-import { fetchAllFormTabs, FullyLoadFormAction } from 'src/app/store/form/form.actions';
-import { AppState } from 'src/app/store/store.module';
-import { DisplayedNote, SectionsState }  from '../answers.model';
-import { getSelectedAnswersAsObject, getSelectedAnswersLoadingStatus, getSpecificThreadByObserver } from '../../store/answer/answer.selectors';
-import { getFormItems, getFullyLoadedForms } from '../../store/form/form.selectors';
-
-import { getNotes, getNotesAsObject, getNotesLoadingStatus, getNotesMergedWithQuestions } from '../../store/note/note.selectors';
-import { Note } from 'src/app/models/note.model';
-import { LoadNotesAction } from 'src/app/store/note/note.actions';
-import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
-
-
-const notesTab = { id: -1, description: 'Notes' };
+import {Component, Inject, OnInit, TemplateRef, ViewChild} from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import {Store} from '@ngrx/store';
+import {TranslateService} from '@ngx-translate/core';
+import {combineLatest, concat, merge, Observable, of, Subject} from 'rxjs';
+import {
+  debounceTime,
+  delay,
+  distinctUntilChanged,
+  distinctUntilKeyChanged,
+  filter,
+  map,
+  scan,
+  shareReplay,
+  startWith,
+  take,
+  tap
+} from 'rxjs/operators';
+import {AnswerExtra} from 'src/app/models/answer.extra.model';
+import {AnswerThread} from 'src/app/models/answer.thread.model';
+import {FormDetails} from 'src/app/models/form.info.model';
+import {Form} from 'src/app/models/form.model';
+import {BASE_BUTTON_VARIANTS, Variants} from 'src/app/shared/base-button/base-button.component';
+import {fetchAllFormTabs, FullyLoadFormAction} from 'src/app/store/form/form.actions';
+import {AppState} from 'src/app/store/store.module';
+import {getSelectedAnswersAsObject, getSpecificThreadByObserver} from '../../store/answer/answer.selectors';
+import {getFormItems, getFullyLoadedForms} from '../../store/form/form.selectors';
+import {getNotesAsObject, getNotesMergedWithQuestions} from '../../store/note/note.selectors';
+import {LoadNotesAction} from 'src/app/store/note/note.actions';
+import {NgbModal} from '@ng-bootstrap/ng-bootstrap';
+
+
+const notesTab = {id: -1, description: 'Notes'};
 
 type Tab = Partial<FormDetails>
 
 @Component({
-  selector: 'answer-details',
+  selector: 'app-answer-details',
```

Comment:
This file contains auto-formatting and this line change ^^^ according to TSlint.
CC @Andrei0872